src/rgw/rgw_reshard.cc

```diff
@@ -888,18 +888,17 @@ int RGWReshard::list(int logshard_num, string& marker, uint32_t max, std::list<c
 
   int ret = cls_rgw_reshard_list(store->getRados()->reshard_pool_ctx, logshard_oid, marker, max, entries, is_truncated);
 
-  if (ret < 0) {
-    lderr(store->ctx()) << "ERROR: failed to list reshard log entries, oid=" << logshard_oid << " " 
-    << "marker=" << marker << " " << cpp_strerror(ret) << dendl;
-    if (ret == -ENOENT) {
-      *is_truncated = false;
-      ret = 0;
-    } else {
-      if (ret == -EACCES) {
-        lderr(store->ctx()) << "access denied to pool " << store->svc()->zone->get_zone_params().reshard_pool
-                          << ". Fix the pool access permissions of your client" << dendl;
-      }
-    } 
+  if (ret == -ENOENT) {
+    // these shard objects aren't created until we actually write something to
+    // them, so treat ENOENT as a successful empty listing
+    *is_truncated = false;
+    ret = 0;
+  } else if (ret == -EACCES) {
+    lderr(store->ctx) << "ERROR: access denied to pool " << store->svc()->zone->get_zone_params().reshard_pool
+                      << ". Fix the pool access permissions of your client" << dendl;
+  } else if (ret < 0) {
+    lderr(store->ctx) << "ERROR: failed to list reshard log entries, oid="
+        << logshard_oid << " marker=" << marker << " " << cpp_strerror(ret) << dendl;
```

Comment:
`ctx` is a member function, so this should be `store->ctx()` like the code it replaces

build log:
```
src/rgw/rgw_reshard.cc: In member function ‘int RGWReshard::list(int, std::__cxx11::string&, uint32_t, std::__cxx11::list<cls_rgw_reshard_entry>&, bool*)’:
src/common/dout.h:159:8: error: invalid use of non-static member function ‘virtual ceph::common::CephContext* rgw::sal::RGWRadosStore::ctx()’
   }(cct);        \
        ^
src/common/dout.h:177:20: note: in expansion of macro ‘dout_impl’
 #define lderr(cct) dout_impl(cct, ceph_subsys_, -1) dout_prefix
                    ^~~~~~~~~
src/rgw/rgw_reshard.cc:904:5: note: in expansion of macro ‘lderr’
     lderr(store->ctx) << "ERROR: access denied to pool " << store->svc()->zone->get_zone_params().reshard_pool
     ^~~~~
In file included from src/rgw/rgw_reshard.cc:11:
src/rgw/rgw_sal_rados.h:328:26: note: declared here
     virtual CephContext *ctx(void) { return rados->ctx(); }
                          ^~~
```